Code warnings in byteswap.c

```
[ 26%] Building C object src/CMakeFiles/bacio_4_c.dir/byteswap.c.o
cd /home/ed/NCEPLIBS-bacio/b/src && /usr/bin/cc -DLINUX -DUNDERSCORE  -g -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -fsanitize=address -Wall -MD -MT src/CMakeFiles/bacio_4_c.dir/byteswap.c.o -MF CMakeFiles/bacio_4_c.dir/byteswap.c.o.d -o CMakeFiles/bacio_4_c.dir/byteswap.c.o -c /home/ed/NCEPLIBS-bacio/src/byteswap.c
/home/ed/NCEPLIBS-bacio/src/byteswap.c:33: warning: "LINUX" redefined
   33 | #define LINUX /**< Define LINUX. */
      | 
<command-line>: note: this is the location of the previous definition
/home/ed/NCEPLIBS-bacio/src/byteswap.c: In function ‘simple_swap_64’:
/home/ed/NCEPLIBS-bacio/src/byteswap.c:58:21: warning: suggest parentheses around comparison in operand of ‘&’ [-Wparentheses]
   58 |   if( ((size_t)data)&0x5 != 0 ) {
      |                     ^
/home/ed/NCEPLIBS-bacio/src/byteswap.c: In function ‘simple_swap_32’:
/home/ed/NCEPLIBS-bacio/src/byteswap.c:88:21: warning: suggest parentheses around comparison in operand of ‘&’ [-Wparentheses]
   88 |   if( ((size_t)data)&0x3 != 0 ) {
      |                     ^
/home/ed/NCEPLIBS-bacio/src/byteswap.c: In function ‘simple_swap_16’:
/home/ed/NCEPLIBS-bacio/src/byteswap.c:114:21: warning: suggest parentheses around comparison in operand of ‘&’ [-Wparentheses]
  114 |   if( ((size_t)data)&0x1 != 0 ) {
      |                     ^
/home/ed/NCEPLIBS-bacio/src/byteswap.c: In function ‘macro_swap_64’:
/home/ed/NCEPLIBS-bacio/src/byteswap.c:139:21: warning: suggest parentheses around comparison in operand of ‘&’ [-Wparentheses]
  139 |   if( ((size_t)data)&0x5 != 0 ) {
      |                     ^
/home/ed/NCEPLIBS-bacio/src/byteswap.c: In function ‘macro_swap_32’:
/home/ed/NCEPLIBS-bacio/src/byteswap.c:162:21: warning: suggest parentheses around comparison in operand of ‘&’ [-Wparentheses]
  162 |   if( ((size_t)data)&0x3 != 0 ) {
      |                     ^
/home/ed/NCEPLIBS-bacio/src/byteswap.c: In function ‘macro_swap_16’:
/home/ed/NCEPLIBS-bacio/src/byteswap.c:185:21: warning: suggest parentheses around comparison in operand of ‘&’ [-Wparentheses]
  185 |   if( ((size_t)data)&0x1 != 0 ) {
      |                     ^
/home/ed/NCEPLIBS-bacio/src/byteswap.c: In function ‘block_macro_swap_32’:
/home/ed/NCEPLIBS-bacio/src/byteswap.c:208:21: warning: suggest parentheses around comparison in operand of ‘&’ [-Wparentheses]
  208 |   if( ((size_t)data)&0x3 != 0 ) {
      |                     ^
/home/ed/NCEPLIBS-bacio/src/byteswap.c: In function ‘block_macro_swap_16’:
/home/ed/NCEPLIBS-bacio/src/byteswap.c:237:21: warning: suggest parentheses around comparison in operand of ‘&’ [-Wparentheses]
  237 |   if( ((size_t)data)&0x1 != 0 ) {
      |                     ^
/home/ed/NCEPLIBS-bacio/src/byteswap.c: In function ‘block_macro_swap_64’:
/home/ed/NCEPLIBS-bacio/src/byteswap.c:266:21: warning: suggest parentheses around comparison in operand of ‘&’ [-Wparentheses]
  266 |   if( ((size_t)data)&0x5 != 0 ) {
      |                     ^
At top level:
/home/ed/NCEPLIBS-bacio/src/byteswap.c:263:12: warning: ‘block_macro_swap_64’ defined but not used [-Wunused-function]
  263 | static int block_macro_swap_64(void *data,size_t len) {
      |            ^~~~~~~~~~~~~~~~~~~
/home/ed/NCEPLIBS-bacio/src/byteswap.c:234:12: warning: ‘block_macro_swap_16’ defined but not used [-Wunused-function]
  234 | static int block_macro_swap_16(void *data,size_t len) {
      |            ^~~~~~~~~~~~~~~~~~~
/home/ed/NCEPLIBS-bacio/src/byteswap.c:205:12: warning: ‘block_macro_swap_32’ defined but not used [-Wunused-function]
  205 | static int block_macro_swap_32(void *data,size_t len) {
      |            ^~~~~~~~~~~~~~~~~~~
/home/ed/NCEPLIBS-bacio/src/byteswap.c:182:12: warning: ‘macro_swap_16’ defined but not used [-Wunused-function]
  182 | static int macro_swap_16(void *data,size_t len) {
      |            ^~~~~~~~~~~~~
/home/ed/NCEPLIBS-bacio/src/byteswap.c:159:12: warning: ‘macro_swap_32’ defined but not used [-Wunused-function]
  159 | static int macro_swap_32(void *data,size_t len) {
      |            ^~~~~~~~~~~~~
/home/ed/NCEPLIBS-bacio/src/byteswap.c:55:12: warning: ‘simple_swap_64’ defined but not used [-Wunused-function]
   55 | static int simple_swap_64(void *data,size_t len) {
      |            ^~~~~~~~~~~~~~
make[2]: Leaving directory '/home/ed/NCEPLIBS-bacio/b'

```

--- a/src/bacio.c
+++ b/src/bacio.c
@@ -102,9 +102,8 @@ bacio_ (int * mode, int * start, int *newpos, int * size, int * no,
         int * nactual, int * fdes, const char *fname, char *datary,
         int  namelen, int  datanamelen)
 {
-    int i, j, jret, seekret;
+    int i, jret, seekret;
     char *realname;
-    int tcharval;
     size_t count;
 
     /* Initialization(s) */
@@ -393,9 +392,8 @@ baciol_(int * mode, long int * start, long int *newpos, int * size, long int * n
         long int * nactual, int * fdes, const char *fname, char *datary,
         int  namelen, int  datanamelen)
 {
-    int i, j, jret, seekret;
+    int i, jret, seekret;
     char *realname;
-    int tcharval;
     size_t count;
 
     /* Initialization(s) */

--- a/src/byteswap.c
+++ b/src/byteswap.c
@@ -55,7 +55,7 @@ simple_swap_32(void *data,size_t len)
 {
     size_t i;
     uint32_t *udata;
-    if( ((size_t)data)&0x3 != 0 ) {
+    if( (((size_t)data)&0x3) != 0 ) {
         if (send_errors)
             fprintf(stderr,"ERROR: pointer to 32-bit integer is not 32-bit aligned (pointer is 0x%llx)\n",(long long)data);
         return 0;
@@ -83,7 +83,7 @@ simple_swap_16(void *data,size_t len)
 {
     size_t i;
     uint16_t *udata;
-    if( ((size_t)data)&0x1 != 0 ) {
+    if( (((size_t)data)&0x1) != 0 ) {
         if (send_errors)
             fprintf(stderr,"ERROR: pointer to 16-bit integer is not 16-bit aligned (pointer is 0x%llx)\n",(long long)data);
         return 0;
@@ -110,7 +110,7 @@ macro_swap_64(void *data,size_t len)
 {
     size_t i;
     uint64_t *udata;
-    if( ((size_t)data)&0x5 != 0 ) {
+    if ((((size_t)data)&0x5) != 0) {
         if (send_errors)
             fprintf(stderr,"ERROR: pointer to 64-bit integer is not 64-bit aligned (pointer is 0x%llx)\n",(long long)data);
         return 0;

--- a/src/byteswap.c
+++ b/src/byteswap.c
@@ -55,7 +55,7 @@ simple_swap_32(void *data,size_t len)
 {
     size_t i;
     uint32_t *udata;
-    if (((size_t)data & 0x3) != 0)
+    if ((size_t)data & 0x3)
     {
         if (send_errors)
             fprintf(stderr,"ERROR: pointer to 32-bit integer is not 32-bit aligned (pointer is 0x%llx)\n",(long long)data);
@@ -84,7 +84,7 @@ simple_swap_16(void *data,size_t len)
 {
     size_t i;
     uint16_t *udata;
-    if (((size_t)data & 0x1) != 0)
+    if ((size_t)data & 0x1)
     {
         if (send_errors)
             fprintf(stderr,"ERROR: pointer to 16-bit integer is not 16-bit aligned (pointer is 0x%llx)\n",(long long)data);
@@ -112,7 +112,7 @@ macro_swap_64(void *data,size_t len)
 {
     size_t i;
     uint64_t *udata;
-    if (((size_t)data & 0x5) != 0)
+    if ((size_t)data & 0x5)
     {
         if (send_errors)
             fprintf(stderr,"ERROR: pointer to 64-bit integer is not 64-bit aligned (pointer is 0x%llx)\n",(long long)data);

--- a/src/byteswap.c
+++ b/src/byteswap.c
@@ -55,7 +55,8 @@ simple_swap_32(void *data,size_t len)
 {
     size_t i;
     uint32_t *udata;
-    if( (((size_t)data)&0x3) != 0 ) {
+    if (((size_t)data & 0x3) != 0)
+    {
         if (send_errors)
             fprintf(stderr,"ERROR: pointer to 32-bit integer is not 32-bit aligned (pointer is 0x%llx)\n",(long long)data);
         return 0;
@@ -83,7 +84,8 @@ simple_swap_16(void *data,size_t len)
 {
     size_t i;
     uint16_t *udata;
-    if( (((size_t)data)&0x1) != 0 ) {
+    if (((size_t)data & 0x1) != 0)
+    {
         if (send_errors)
             fprintf(stderr,"ERROR: pointer to 16-bit integer is not 16-bit aligned (pointer is 0x%llx)\n",(long long)data);
         return 0;
@@ -110,7 +112,8 @@ macro_swap_64(void *data,size_t len)
 {
     size_t i;
     uint64_t *udata;
-    if ((((size_t)data)&0x5) != 0) {
+    if (((size_t)data & 0x5) != 0)
+    {
         if (send_errors)
             fprintf(stderr,"ERROR: pointer to 64-bit integer is not 64-bit aligned (pointer is 0x%llx)\n",(long long)data);
         return 0;

